### PR TITLE
🐛 Source TikTok: fix default filters params - include deleted ad_groups, ad, campaigns to reports 

### PR DESCRIPTION
**Pull request by efimmatytsin**
## What
We employ the Airbyte TikTok Marketing connector with the purpose of capturing our marketing data. Although its functionality is satisfactory, a significant issue has arisen due to the lack of built-in filters. By default, in the absence of any filter applications, the ad_status [ field automatically defaults to STATUS_NOT_DELETE](https://ads.tiktok.com/marketing_api/docs?id=1751443975608321) within the synchronous basic reports. This has resulted in substantial data loss, as a large number of our ads have been deleted. Therefore, we need to integrate a data filtering ability.

## How
Include STATUS_ALL filter for ad_status, adgroup_status, campaign_status -  to every report request.

## 🚨 User Impact 🚨
Numbers in reports can be different, because they will include deleted ad, adgroup, campaign statistics.

I can't run integration tests on my local machine, but tested it with 
```
python main.py read --config secrets/config.json --catalog integration_tests/configured_catalog.json
```

Any help or comments will be appreciated.

### CODE REVIEW

**Review by archangelic — CHANGES_REQUESTED** [review]
Thanks so much for your contribution! 

I have one small change below. 

You will need to increment the version following this: [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors)

Also, in `unit_tests/test_streams.py` at line 210, you will need to insert this into the assert statement:
`"filters": '[{"filter_value": ["STATUS_ALL"], "field_name": "ad_status", "filter_type": "IN"}, {"filter_value": ["STATUS_ALL"], "field_name": "campaign_status", "filter_type": "IN"}, {"filter_value": ["STATUS_ALL"], "field_name": "adgroup_status", "filter_type": "IN"}]',`

Let me know if you have any questions!

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py [hunk 1]
```diff
@@ -510,6 +509,26 @@ def __init__(self, **kwargs):
...
+            {
+                "filter_value": ["STATUS_ALL"],
+                "field_name": "campaign_status",
+                "filter_type": "IN"
+            },
+            {
+                "filter_value": ["STATUS_ALL"],
+                "field_name": "adgroup_status",
+                "filter_type": "IN"
+            },
```
[the remark this answers is not in the document]
Comment (efimmatytsin) [reply airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py:529]: Thank you! What formatter are you using?

[the remark this answers is not in the document]
Comment (efimmatytsin) [reply airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py]: also I append what you asked to 210 line, but I can't test it  locally 
image
it's ok?

[the remark this answers is not in the document]
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py]: Looks 2 unit tests are using the integration test sandbox... so it isn't possible to finish the unit test locally without them.

[the remark this answers is not in the document]
Comment (archangelic) [reply airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py]: > Thank you! What formatter are you using?

We have information about running the linter here: https://docs.airbyte.com/contributing-to-airbyte/resources/developing-locally/#connector

And yes, you won't be able to run the tests, but the change I suggested will let them run. Thanks for doing that!

Can you also iterate the version in `metadata.yaml`?

[the remark this answers is not in the document]
Comment (efimmatytsin) [reply airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py]: Hello @archangelic ! done

**Comment by archangelic** [comment]
It looks like there is a schema change that needs to be made before this can be merged. I've opened #27988 to resolve that. Once that PR is merged, you will need to merge the changes and then we will move forward with getting this finished!

**Comment by archangelic** [comment]
@efimmatytsin The PR was merged. Can you merge the changes from master branch? You will need to update the version to `3.3.1`.

**Comment by efimmatytsin** [comment]
@archangelic done!

**Review by archangelic — APPROVED** [review]

**Comment by archangelic** [comment]
I'm going to initiate tests on my end and if all's well, we will merge!

**Comment by archangelic** [comment]
Can you run the linter and push the changes? https://docs.airbyte.com/contributing-to-airbyte/resources/developing-locally/#connector

**Comment by efimmatytsin** [comment]
@archangelic Hello! done!

**Comment by efimmatytsin** [comment]
@archangelic Hello! Can I do something?

**Comment by archangelic** [comment]
Hey! Everything is good on your end, but this needs further review before it can be merged. That should happen soon!

**Comment by archangelic** [comment]
@efimmatytsin It looks like there is only one thing missing, and that is a changelog entry in `airbyte/docs/integrations/sources/tiktok-marketing.md`. Once you add that, we should be set! Thanks for your patience during this process!

**Comment by efimmatytsin** [comment]
@archangelic Thank you! Done.

**Review by davydov-d — CHANGES_REQUESTED** [review]
@efimmatytsin thanks for your effort! Overall, PR looks good, but regarding this change affects the amount of data that would be emitted, I'd suggest you make this filter optional, let's say as a part of the input config with default value equal to the one that's used now. Also, please bump minor version instead of a patch. Thanks!

**Comment by efimmatytsin** [comment]
Hello @davydov-d ! Thank you for your comment! Can you please describe how do you think this config should looks like?One for each report or just one for all reports?

**Comment by davydov-d** [comment]
> Hello @davydov-d ! Thank you for your comment! Can you please describe how do you think this config should looks like?One for each report or just one for all reports?

@efimmatytsin I think we should add a boolean field, for instance
```
...
"properties": {
      ...
      "include_deleted": {
        "title": "Include Deleted Data in Reports",
        "description": "Set to active if you want to include deleted data in reports.",
        "default": false,
        "type": "boolean"
      },
      ...
}      
```

adding one option per report would be an overengineering at the moment, I guess we can scale it when we need it, so let's start with one field for all reports

**Comment by efimmatytsin** [comment]
@davydov-d Thank you!

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py [hunk 2]
```diff
@@ -502,6 +503,7 @@ def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
     def __init__(self, **kwargs):
         report_granularity = kwargs.pop("report_granularity", None)
         self.attribution_window = kwargs.get("attribution_window") or 0
+        self.include_deleted = kwargs.get("include_deleted") or False
```
[the remark this answers is not in the document]
Comment (efimmatytsin) [reply airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py:506]: sure :D

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/spec.json [hunk 3]
```diff
@@ -91,6 +91,12 @@
         "type": "string",
         "format": "date"
       },
+      "include_deleted": {
+        "title": "Include Deleted Data in Reports",
+        "description": "Set to active if you want to include deleted data in reports.",
+        "default": false,
+        "type": "boolean"
+      },
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/spec.json:99]: please add `order` param

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py [hunk 4]
```diff
@@ -482,6 +482,7 @@ class BasicReports(IncrementalTiktokStream, ABC):
 
     schema_name = "basic_reports"
     report_granularity = None
+    include_deleted = False
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py:485]:
```suggestion
```
No need to define a class wide variable if you're defining an instance wide variable

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py [hunk 5]
```diff
@@ -502,6 +503,7 @@ def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
     def __init__(self, **kwargs):
         report_granularity = kwargs.pop("report_granularity", None)
         self.attribution_window = kwargs.get("attribution_window") or 0
+        self.include_deleted = kwargs.get("include_deleted", False)
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py:506]: looks like we can never get `True` here, please see https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/source.py#L68, another change should be added there to pass new config variables further

**Review by davydov-d — CHANGES_REQUESTED** [review]
we're getting closer! an important thing is to correctly pass config variables to streams, please see comments

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py (hunk 4 above)
Comment (efimmatytsin) [reply airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py:485]: done.

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py (hunk 5 above)
Comment (efimmatytsin) [reply airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py:506]: done

**Comment by davydov-d** [comment]
@efimmatytsin there's been a dependency upgrade which causes a build failure. I'd like to ask you to add one more change like [this one](https://github.com/airbytehq/airbyte/commit/7be3c45b2c17367cd73967321f5018ef5eee084b) to overcome it

**Comment by efimmatytsin** [comment]
hello @davydov-d thank you for your help! I made local build and it was successfully, I can do that but can you please recheck? Branch was updated.
image

**Comment by davydov-d** [comment]
@efimmatytsin we have two tests failing in the acceptance test suit, looks like it's not related to your changes but still it's blocking us from merging this PR. I'm working on figuring out the cause and fixing it. Will keep you updated

**Comment by efimmatytsin** [comment]
@davydov-d Thank you so much!

**Comment by efimmatytsin** [comment]
Hello @davydov-d any updates here?

**Comment by davydov-d** [comment]
@efimmatytsin unfortunately no updates yet. We're getting even more errors due to TikTok server errors :(

**Comment by davydov-d** [comment]
@efimmatytsin please rebase from the `ddavydov/tests-for-tiktok-SCNTALS-1949` branch, I've rebased from master and resolved latest conflicts - all the checks pass!

**Comment by davydov-d** [comment]
> @efimmatytsin please rebase from the `ddavydov/tests-for-tiktok-SCNTALS-1949` branch, I've rebased from master and resolved latest conflicts - all the checks pass!

or you can achieve the same result by rebasing from master and resolving conflicts manually

**Comment by efimmatytsin** [comment]
Hello @davydov-d ! Done.

**Review by davydov-d — APPROVED** [review]

**Comment by davydov-d** [comment]
@marcosmarxm we need your approve as well please as you have requested changes

**Review by marcosmarxm — APPROVED** [review]
LGTM Thanks @efimmatytsin

**Pull request merged**